python/triton/compiler/compiler.py

```diff
@@ -397,7 +402,7 @@ def _init_handles(self):
         # create launcher
         self.run = driver.active.launcher_cls(self.src, self.metadata)
         # not enough shared memory to run the kernel
-        max_shared = driver.active.utils.get_device_properties(device)["max_shared_mem"]
+        max_shared = max_shared_mem(device)
```

Comment:
Can we memorize it in driver.py instead of here?

Reply:
I think so, but it'll be a bigger change-- I think we'll need to refactor DriverBase because currently each backend sets this method from a natively compiled function, e.g. https://github.com/triton-lang/triton/blob/main/third_party/amd/backend/driver.py#L162

Maybe we can rename  the function each backend needs to implement from`get_device_properties` > `_ get_device_properties` and then have an implementation of `get_device_properties` which memoizes and calls `_ get_device_properties`. If I have a moment later today I'll put up an RFC with that change so we can chat over actual code